version: '3'
services:
  django:
    build:
      context: .
      dockerfile: ./dev/django.Dockerfile
    command: ["./manage.py", "runserver", "0.0.0.0:8000"]
    # Log printing via Rich is enhanced by a TTY
    tty: true
    devices:
      - /dev/fuse:/dev/fuse
    security_opt:
      - apparmor:unconfined
    cap_add:
      - SYS_ADMIN
    env_file: ./dev/.env.docker-compose
    volumes:
      - .:/opt/django-project
    ports:
      - 8000:8000
    depends_on:
      - postgres
      - rabbitmq
      - minio

  celery:
    build:
      context: .
      dockerfile: ./dev/django.Dockerfile
    command: [
      "celery",
      "--app", "atlascope.celery",
      "worker",
      "--loglevel", "INFO",
      "--without-heartbeat"
    ]
    # Docker Compose does not set the TTY width, which causes Celery errors
    tty: false
    devices:
      - /dev/fuse:/dev/fuse
    security_opt:
      - apparmor:unconfined
    cap_add:
      - SYS_ADMIN
    env_file: ./dev/.env.docker-compose
    volumes:
      - .:/opt/django-project
    depends_on:
      - postgres
      - rabbitmq
      - minio

  demotiles:
    build:
      context: ./dev/demotiles
      dockerfile: ./Dockerfile
    ports:
      - 9005:9005
